invenio_oaiserver/receivers.py

```diff
@@ -112,11 +112,16 @@ def __call__(self, record, **kwargs):
 
         :param record: The record data.
         """
-        record.setdefault('_oai', {})
-        record['_oai'].update({
-            'sets': get_record_sets(record=record, matcher=self.matcher),
-            'updated': datetime_to_datestamp(datetime.utcnow()),
-        })
+        if '_oai' in record and 'id' in record['_oai']:
+            old_sets = sorted(record['_oai'].get('sets', []))
```

Comment:
That varies on the size of x and y, for ~20-30 elements, ``sorted`` seems to be faster